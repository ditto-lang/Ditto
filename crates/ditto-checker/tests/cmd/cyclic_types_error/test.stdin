module Test exports (..)

type A = A(B)
type B(c) = B(A, c)
