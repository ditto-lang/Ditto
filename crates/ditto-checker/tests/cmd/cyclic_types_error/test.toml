bin.name = "ditto-checker-testbin-check-module"
args = ["cyclic_types_error.ditto"]
fs.sandbox = true
status = "failed"
