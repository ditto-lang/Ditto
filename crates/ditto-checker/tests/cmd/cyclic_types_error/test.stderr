
  × kinds don't unify
   ╭─[cyclic_types_error.ditto:2:1]
 2 │ 
 3 │ type A = A(B)
   ·            ┬
   ·            ╰── here
 4 │ type B(c) = B(A, c)
   ╰────
  help: expected Type
        got ($1) -> Type

